### DISCUSSION TITLE
Autoloading of enum display names and translations

### DISCUSSION THREAD

**Discussion by ldommer** [filamentphp/filament#14492] [Ideas]
@danharrin

I recently used Filament for the first time in a project and am blown away by the amount of features and the structure and thoughtfulness of the framework. Great work!

I'm a big fan of enums and use them a lot where appropriate. What I find helpful is the feature of autoloading their display names by simply implementing the `HasLabel` interface and a `getLabel` method in an enum, returning a string for their matching case.

But I thought this feature could be taken a step further, providing the following benefits:

- reduce the amount of unnecessary code (picking up a Filament key concept - as less code as possible)
- emphasize convention before configuration (picking up a Laravel key concept - as much convention as possible)
- optimize code structure (no strings / translations should be located in a PHP class / enum and enum translations might also be organized in translation files anyway)
- provide easy auto-translation of enums into different languages by using the Laravel localization system
- reduce possible string duplications (if enum translations might also be organized in translation files, these are either redundant to the hardcoded ones in the enum or the code loading the enum translations from the translation files in the `getLabel` method could easily be abstracted away)

So I would like to contribute a `Translatable` trait for enums, that provides autoloading of enum translations from Laravel translation files by simply implementing it in an enum.

For example:

```php
<?php

namespace App\Enums;

use App\Enums\Traits\Enumerable;
use Filament\Support\Contracts\HasLabel;

enum PriceType: string implements HasLabel
{
    use Translatable;

    case DailyRate = 'daily_rate';
    case FixedPrice = 'fixed_price';
    case HourlyRate = 'hourly_rate';
    case PriceByValue = 'price_by_value';
}
```

```php
<?php

return [
    'daily_rate'     => 'Tagessatz',
    'fixed_price'    => 'Festpreis',
    'hourly_rate'    => 'Stundensatz',
    'price_by_value' => 'Preis nach Wert',
];
```

This is enough to provide the benefits I discussed above, but also keeps the possibility of organizing enum display names in the old way. Is the contribution of a `Translatable` trait doing this suitable for the framework?

**Comment by danharrin** [comment]
Hi, thanks for your kind words! However, I don't think this is something I would include in the framework, as it is quite opinionated (I don't structure my translation files like that, for example) and is something you could have in your project, or something that you could release in a package.
